### PR TITLE
Feature/iwyu part 4

### PR DESCRIPTION
**Pull request by petersteneteg**
Changes proposed in this PR:
 * Include cleanup for a number of modules

This should be the last of the core modules

### CODE REVIEW

**Review by martinfalk — DISMISSED** [review]

**Comment by martinfalk** [comment]
Include removal of `QMenu` was a bit too aggressive.
```
Error: D:\a\inviwo\inviwo\inviwo\modules\qtwidgets\include\modules/qtwidgets/properties/anglepropertywidgetqt.h(117,33): error C2027: use of undefined type 'QMenu' [D:\a\inviwo\inviwo\build\modules\qtwidgets\inviwo-module-qtwidgets.vcxproj]
```

```
Warning: /home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/include/modules/qtwidgets/properties/ordinalminmaxpropertywidgetqt.h:291:31: warning: invalid use of incomplete type ‘class QMenu’
  291 |     auto settingsAction = menu->addAction(tr("&Property settings..."));
      |                               ^~
In file included from /home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/include/modules/qtwidgets/properties/ordinalminmaxpropertywidgetqt.h:37,
                 from /home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/src/properties/ordinalminmaxpropertywidgetqt.cpp:30:
/home/runner/work/inviwo/inviwo/inviwo/modules/qtwidgets/include/modules/qtwidgets/inviwoqtutils.h:66:7: note: forward declaration of ‘class QMenu’
   66 | class QMenu;
      |       ^~~~~
```

File: modules/animationqt/src/animationviewqt.cpp [hunk 1]
```diff
@@ -28,22 +28,44 @@
...
+#include <modules/animation/animationcontroller.h>            // for AnimationController
+#include <modules/animation/animationcontrollerobserver.h>    // for AnimationControllerObservable
+#include <modules/animation/datastructures/animationstate.h>  // for AnimationState
+#include <modules/animation/datastructures/animationtime.h>   // for Seconds
+#include <modules/animationqt/animationeditorqt.h>            // for AnimationEditorQt
+#include <modules/animationqt/widgets/editorconstants.h>      // for FindDivisionsResult, widthP...
+
+#include <cstdlib>  // for abs, size_t
+#include <cmath>    // for floor, abs
+
```
Comment (ResearchDaniel) [inline modules/animationqt/src/animationviewqt.cpp:41]: Add back
include <warn/push>
 #include <warn/ignore/all>

File: modules/animationqt/src/demo/demonavigatordockwidgetqt.cpp [hunk 2]
```diff
@@ -27,31 +27,37 @@
...
+#include <modules/animation/demo/democontrollerobserver.h>       // for DemoControllerObservable
+#include <modules/animationqt/demo/demonavigatordockwidgetqt.h>  // for DemoNavigatorDockWidgetQt
+#include <modules/qtwidgets/inviwodockwidget.h>                  // for InviwoDockWidget
+#include <modules/qtwidgets/inviwoqtutils.h>                     // for emToPx, toQString
+#include <modules/qtwidgets/properties/propertywidgetqt.h>       // for PropertyWidgetQt
+
+#include <memory>  // for unique_ptr
+#include <string>  // for string
+#include <vector>  // for vector
+
```
Comment (ResearchDaniel) [inline modules/animationqt/src/demo/demonavigatordockwidgetqt.cpp:42]: Warn push/pop

File: modules/animationqt/src/sequenceeditor/controlsequenceeditor.cpp [hunk 3]
```diff
@@ -29,25 +29,26 @@
...
+#include <modules/animation/datastructures/controlkeyframe.h>         // for ControlKeyframe
+#include <modules/animation/datastructures/keyframe.h>                // for Keyframe
+#include <modules/animation/datastructures/keyframeobserver.h>        // for KeyframeObserver
+#include <modules/animation/datastructures/keyframesequence.h>        // for KeyframeSequence
+#include <modules/animation/datastructures/track.h>                   // for Track
+#include <modules/animationqt/sequenceeditor/sequenceeditorwidget.h>  // for SequenceEditorWidget
+#include <modules/qtwidgets/inviwoqtutils.h>                          // for toQString
+
+#include <cstddef>  // for size_t
+
```
Comment (ResearchDaniel) [inline modules/animationqt/src/sequenceeditor/controlsequenceeditor.cpp:42]: Warn push/pop

File: modules/animationqt/src/sequenceeditor/keyframeeditorwidget.cpp [hunk 4]
```diff
@@ -28,25 +28,24 @@
...
+#include <modules/animation/datastructures/keyframe.h>                // for Keyframe
+#include <modules/animation/datastructures/propertytrack.h>           // for BasePropertyTrack
+#include <modules/animation/datastructures/track.h>                   // for Track
+#include <modules/animationqt/sequenceeditor/sequenceeditorwidget.h>  // for SequenceEditorWidget
+#include <modules/qtwidgets/editablelabelqt.h>                        // for EditableLabelQt
+#include <modules/qtwidgets/properties/propertywidgetqt.h>            // for PropertyWidgetQt
+
+#include <functional>  // for __base
+
+#include <QComboBox>       // for QComboBox
```
Comment (ResearchDaniel) [inline modules/animationqt/src/sequenceeditor/keyframeeditorwidget.cpp:46]: Warn push/pop

File: modules/animationqt/src/sequenceeditor/propertysequenceeditor.cpp [hunk 5]
```diff
@@ -28,31 +28,49 @@
...
+#include <modules/qtwidgets/properties/propertywidgetqt.h>             // for PropertyWidgetQt
+
+#include <algorithm>         // for none_of
+#include <cstddef>           // for size_t
+#include <functional>        // for __base
+#include <initializer_list>  // for begin, end
+#include <memory>            // for unique_ptr, unique...
+#include <vector>            // for vector
+
+#include <QComboBox>       // for QComboBox
```
Comment (ResearchDaniel) [inline modules/animationqt/src/sequenceeditor/propertysequenceeditor.cpp:62]: Warn push/pop

File: modules/animationqt/src/sequenceeditor/sequenceeditorwidget.cpp [hunk 6]
```diff
@@ -29,24 +29,22 @@
...
+#include <vector>     // for vector
 
-#include <warn/push>
-#include <warn/ignore/all>
-#include <QVBoxLayout>
-#include <QHBoxLayout>
-#include <QLabel>
-#include <QComboBox>
-#include <warn/pop>
+#include <QVBoxLayout>  // for QVBoxLayout
```
Comment (ResearchDaniel) [inline modules/animationqt/src/sequenceeditor/sequenceeditorwidget.cpp:39]: Warn push/pop?

File: modules/animationqt/src/trackcontrolswidgetqt.cpp [hunk 7]
```diff
@@ -28,17 +28,29 @@
...
+#include <modules/animation/datastructures/animationtime.h>  // for Seconds
+#include <modules/animation/datastructures/track.h>          // for Track
+#include <modules/qtwidgets/inviwoqtutils.h>                 // for toQString
+
+#include <algorithm>  // for lower_bound, upper_bound
+#include <chrono>     // for duration
+#include <iterator>   // for prev
+#include <vector>     // for vector
+
+#include <QAction>       // for QAction
```
Comment (ResearchDaniel) [inline modules/animationqt/src/trackcontrolswidgetqt.cpp:43]: Warn push/pop?

File: modules/animationqt/src/widgets/editorconstants.cpp [hunk 8]
```diff
@@ -29,14 +29,15 @@
...
-#include <warn/ignore/all>
-#include <QApplication>
-#include <warn/pop>
+#include <stdlib.h>   // for abs, size_t
+#include <algorithm>  // for max, min
+#include <array>      // for array
+#include <cmath>      // for round, remainder, pow, abs, floor, log10
+#include <limits>     // for numeric_limits
+
+#include <QApplication>  // for QApplication
```
Comment (ResearchDaniel) [inline modules/animationqt/src/widgets/editorconstants.cpp:38]: Warn push/pop

File: modules/animationqt/src/widgets/keyframesequencewidgetqt.cpp [hunk 9]
```diff
@@ -29,22 +29,38 @@
...
-#include <warn/pop>
+#include <modules/animation/datastructures/keyframesequence.h>  // for KeyframeSequence
+#include <modules/animationqt/widgets/editorconstants.h>        // for timeToScenePos, trackHeig...
+#include <modules/animationqt/widgets/keyframewidgetqt.h>       // for KeyframeWidgetQt
+
+#include <algorithm>  // for max
+#include <cstddef>    // for size_t
+#include <utility>    // for pair
+
+#include <QApplication>     // for QApplication
```
Comment (ResearchDaniel) [inline modules/animationqt/src/widgets/keyframesequencewidgetqt.cpp:40]: Warn push/pop

File: modules/animationqt/src/animationeditordockwidgetqt.cpp [hunk 10]
```diff
@@ -29,51 +29,71 @@
...
-#include <QVBoxLayout>
-#include <QHBoxLayout>
-#include <QSplitter>
-#include <QSettings>
-#include <QToolBar>
-#include <QMainWindow>
-#include <QMenuBar>
-#include <QWidget>
-#include <QScrollBar>
-#include <warn/pop>
```
Comment (ResearchDaniel) [inline modules/animationqt/src/animationeditordockwidgetqt.cpp:73]: Warn pop removed, but not push

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Is it intentional that warn/push/pop is removed in many places?

**Comment by petersteneteg** [comment]
yes, the warn pop/push are removed intentionally. It seems most most warnings they were needed for has been fixed upstream

File: modules/animationqt/src/animationeditordockwidgetqt.cpp (hunk 10 above)
Comment (petersteneteg) [reply modules/animationqt/src/animationeditordockwidgetqt.cpp:73]: good catch

**Comment by ResearchDaniel** [comment]
> yes, the warn pop/push are removed intentionally. It seems most most warnings they were needed for has been fixed upstream

Ok. I saw at least one place where the pop was removed, but not the push

File: tools/iwyu/README.md [hunk 11]
```diff
@@ -32,6 +32,30 @@ Example preset to for running iwyu
   ]
 }
 
-
-
+Fixes for qt, modify iwyu to avoid some circular include in global.h see
```
Comment (ResearchDaniel) [inline tools/iwyu/README.md:35]:
```suggestion
Fixes for qt. Modify IWYU (Include What You Use) to avoid some circular include in global.h, see
```

**Review by ResearchDaniel — APPROVED** [review]

**Comment by ResearchDaniel** [comment]
@petersteneteg The following includes are incompatible with Qt5:
```
#include <QEventPoint>      // for QEventPoint, QEventPoi...
#include <QInputDevice>     // for QInputDevice, QInputDe...
#include <QPointingDevice>  // for QPointingDevice
#include <QIODeviceBase>    // for QIODeviceBase::WriteOnly
```

**Comment by petersteneteg** [comment]
Ahh... I forgot about qt5 when doing this. Are you building with qt5 still? I just updated to 6.4, is there any good reason to keep qt5 support?

**Comment by ResearchDaniel** [comment]
Using it on one of my computers still. No special reason except for backwardscompatibility I think.

**Comment by petersteneteg** [comment]
Do you think we should keep supporting it?

**Comment by ResearchDaniel** [comment]
How about bumping version and loosing support?

**Pull request merged**